src/material-experimental/mdc-progress-bar/progress-bar.ts

```diff
@@ -0,0 +1,209 @@
+  selector: 'mat-progress-bar',
+  exportAs: 'matProgressBar',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+    '[attr.aria-valuenow]': '(mode === "indeterminate" || mode === "query") ? null : value',
+    '[attr.mode]': 'mode',
+    'class': 'mat-mdc-progress-bar',
+    '[class._mat-animation-noopable]': '_isNoopAnimation',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-bar.html',
+  styleUrls: ['progress-bar.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatProgressBar extends _MatProgressBarMixinBase implements AfterViewInit, OnDestroy,
+  CanColor {
+
+  constructor(public _elementRef: ElementRef<HTMLElement>,
+              private _ngZone: NgZone,
+              @Optional() private _dir?: Directionality,
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    super(_elementRef);
+    this._isNoopAnimation = _animationMode === 'NoopAnimations';
+    if (_dir) {
+      this._dirChangeSubscription = _dir.change.subscribe(() => this._syncFoundation());
+    }
+  }
+
+  /** Implements all of the logic of the MDC progress bar. */
+  private _foundation: MDCLinearProgressFoundation | undefined;
+
+  /** Adapter used by MDC to interact with the DOM. */
+  private _adapter: MDCLinearProgressAdapter = {
+    addClass: (className: string) => this._rootElement.nativeElement.classList.add(className),
+    getBuffer: () => this._bufferBar.nativeElement,
+    getPrimaryBar: () => this._primaryBar.nativeElement,
+    hasClass: (className: string) => this._rootElement.nativeElement.classList.contains(className),
+    removeClass: (className: string) => this._rootElement.nativeElement.classList.remove(className),
+    setStyle: (el: HTMLElement, styleProperty: string, value: string) => {
+      (el.style as any)[styleProperty] = value;
+    }
+  };
+
+  /** Flag that indicates whether NoopAnimations mode is set to true. */
+  _isNoopAnimation = false;
+
+  /** Value of the progress bar. Defaults to zero. Mirrored to aria-valuenow. */
+  @Input()
+  get value(): number { return this._value; }
+  set value(v: number) {
+    this._value = clamp(v || 0);
+    this._syncFoundation();
+
+    // When noop animation is set to true, trigger animationEnd directly.
+    if (this._isNoopAnimation) {
+      this._emitAnimationEnd();
+    }
+  }
+  private _value = 0;
+
+  /** Buffer value of the progress bar. Defaults to zero. */
+  @Input()
+  get bufferValue(): number { return this._bufferValue || 0; }
+  set bufferValue(v: number) {
+    this._bufferValue = clamp(v || 0);
+    this._syncFoundation();
+  }
+  private _bufferValue = 0;
+
+  @ViewChild('rootElement', {static: false}) _rootElement: ElementRef<HTMLElement>;
+  @ViewChild('primaryBar', {static: false}) _primaryBar: ElementRef<HTMLElement>;
+  @ViewChild('bufferBar', {static: false}) _bufferBar: ElementRef<HTMLElement>;
```

Comment:
Instead of doing ViewChild for these, can we just use `querySelector`? Lately I've been thinking that using `ViewChild` to grab an `ElementRef` is an anti-pattern since it ends up being more costly for no benefit